pyomo/contrib/alternative_solutions/solution.py

```diff
@@ -0,0 +1,158 @@
+
+import json
+import pyomo.environ as pe
+from pyomo.common.collections import ComponentMap, ComponentSet
+from pyomo.contrib.alternative_solutions import aos_utils
+
+
+class Solution:
+    """
+    A class to store solutions from a Pyomo model.
+
+    Attributes
+    ----------
+    variables : ComponentMap
+        A map between Pyomo variables and their values for a solution.
+    fixed_vars : ComponentSet
+        The set of Pyomo variables that are fixed in a solution.
+    objective : ComponentMap
+        A map between Pyomo objectives and their values for a solution.
+
+    Methods
+    -------
+    pprint():
+        Prints a solution.
+    get_variable_name_values(self, ignore_fixed_vars=False):
+        Get a dictionary of variable name-variable value pairs.
+    get_fixed_variable_names(self):
+        Get a list of fixed-variable names.
+    get_objective_name_values(self):
+        Get a dictionary of objective name-objective value pairs.
+    """
+
+    def __init__(self, model, variable_list, include_fixed=True, objective=None):
+        """
+        Constructs a Pyomo Solution object.
+
+        Parameters
+        ----------
+            model : ConcreteModel
+                A concrete Pyomo model.
+            variable_list: A collection of Pyomo _GenereralVarData variables
+                The variables for which the solution will be stored.
+            include_fixed : boolean
+                Boolean indicating that fixed variables should be added to the
+                solution.
+            objective: None or Objective
+                The objective functions for which the value will be saved. None
+                indicates that the active objective should be used, but a
+                different objective can be stored as well.
+        """
+
+        self.variables = ComponentMap()
+        self.fixed_vars = ComponentSet()
+        for var in variable_list:
+            is_fixed = var.is_fixed()
+            if is_fixed:
+                self.fixed_vars.add(var)
+            if include_fixed or not is_fixed:
+                self.variables[var] = pe.value(var)
+
+        if objective is None:
+            objective = aos_utils.get_active_objective(model)
+        self.objective = (objective, pe.value(objective))
+
+    @property
+    def objective_value(self):
+        """
+        Returns
+        -------
+            The value of the objective.
+        """
+        return self.objective[1]
+
+    def pprint(self, round_discrete=True, sort_keys=True, indent=4):
+        """
+        Print the solution variables and objective values.
+
+        Parameters
+        ----------
+            rounded_discrete : boolean
+                If True, then round discrete variable values before printing.
+        """
+        print(
+            self.to_string(
+                round_discrete=round_discrete, sort_keys=sort_keys, indent=indent
+            )
+        )  # pragma: no cover
```

Comment:
This should be hard to test, should it? Or honestly leaving it uncovered for now is probably fine.

Reply:
Capturing I/O can be tricky.

Reply:
```python
with pyomo.common.tee.capture_output() as OUT:
   # run test
self.assertEqual(OUT.getvalue(), reference_string)
```
If you have issues with the thing you are testing printing directly to the underlying C file descriptors (some solvers do this), use `capture_output(capture_fd=True)`